cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -232,7 +232,14 @@ getTxId (ShelleyTxBody era tx _) =
 --
 
 data TxIn = TxIn TxId TxIx
-  deriving (Eq, Generic, Ord, Show)
+  deriving (Eq, Ord, Show)
+
+instance ToJSON TxIn where
+  toJSON (TxIn txId (TxIx ix)) =
+    Aeson.String ( Text.decodeUtf8 (serialiseToRawBytesHex txId)
+                 <> "#"
+                 <> Text.pack (show ix)
+                 )
```

Comment:
Oh what happened to the `ToJSONKey` instance? Perhaps I misread and you only have the `ToJSONKey` for the underlying Shelley type.

We should provide it for the surface type because it's directly useful and so we can use the simple instance for UTxO above.

The `ToJSONKey` instance should use this rendering as a single string with the #, but the `ToJSON` should still give us an object with two fields`txid` and `txix`.